pkg/controller/dnsendpoint/dnsendpoint_controller.go

```diff
@@ -59,45 +58,44 @@ func Add(mgr manager.Manager) error {
 		return err
 	}
 
-	for _, changeNotifier := range nameServerChangeNotifiers {
-		err := ctrl.Watch(&source.Channel{Source: changeNotifier}, &handler.EnqueueRequestForObject{})
-		if err != nil {
-			log.WithField("controller", controllerName).WithError(err).Error("unable to set up watch for name server changes")
-			return err
-		}
+	if err := ctrl.Watch(&source.Channel{Source: nameServerChangeNotifier}, &handler.EnqueueRequestForObject{}); err != nil {
+		log.WithField("controller", controllerName).WithError(err).Error("unable to set up watch for name server changes")
+		return err
 	}
 
 	return nil
 }
 
-func newReconciler(mgr manager.Manager, kubeClient client.Client) (*ReconcileDNSEndpoint, []chan event.GenericEvent, error) {
-	nameServerScrapers := map[string]*nameServerScraper{}
-	nameServerQueryers := map[string]nameserver.Query{}
-	nameServerChangeNotifiers := []chan event.GenericEvent{}
+type nameServerTool struct {
+	scraper     *nameServerScraper
+	queryClient nameserver.Query
+}
+
+func newReconciler(mgr manager.Manager, kubeClient client.Client) (*ReconcileDNSEndpoint, chan event.GenericEvent, error) {
+	nameServerChangeNotifier := make(chan event.GenericEvent, 1024)
```

Comment:
Don't need to create this until the end of this function. The function does not need to return a non-nil notifier when it returns a nil reconciler.